addons/payment_ogone/models/payment_transaction.py

```diff
@@ -219,10 +243,12 @@ def _process_notification_data(self, notification_data):
         elif payment_status in const.PAYMENT_STATUS_MAPPING['cancel']:
             self._set_canceled()
         elif payment_status in const.PAYMENT_STATUS_MAPPING['declined']:
-            if notification_data.get("NCERRORPLUS"):
-                reason = notification_data.get("NCERRORPLUS")
-            elif notification_data.get("NCERROR"):
-                reason = "Error code: %s" % notification_data.get("NCERROR")
+            if ncerror := notification_data.get('NCERROR'):
+                reason = (
+                    const.PAYMENT_ERROR_TYPES_MAPPING.get(int(ncerror))
+                    or notification_data.get('NCERRORPLUS')
+                    or f"Error code: {ncerror}"
```

Comment:
```suggestion
                    or _("Error code: %s"; ncerror")
```
With the export of the translatable terms.

Reply:
Same for the other reason below.